dev-tools/gen-windows-sys-binding/src/main.rs

```diff
@@ -49,16 +50,39 @@ fn main() -> io::Result<()> {
     // Generate bindings.
     windows_bindgen::bindgen(&args).expect("running bindgen failed");
 
-    let mut bindings = String::new();
-    fs::File::open(temp_file.path())
-        .expect("failed to open temp windows_sys.rs")
-        .read_to_string(&mut bindings)
-        .expect("failed to read temp windows_sys.rs");
+    let bindings =
+        fs::read_to_string(temp_file.path()).expect("failed to read temp windows_sys.rs");
 
     let mut f = fs::File::create(format!("{manifest_dir}/../../src/windows/windows_sys.rs"))
+        .map(BufWriter::new)
         .expect("failed to create windows_sys.rs");
-    f.write_all(PRELUDE.as_bytes())?;
-    f.write_all(bindings.as_bytes())?;
 
-    Ok(())
+    write!(&mut f, "{PRELUDE}\n{bindings}\n").unwrap();
+
+    let mut dll_names: Vec<&str> = Regex::new(r#"link!\("(.*)\.dll""#)
+        .unwrap()
+        .captures_iter(&bindings)
+        .map(|caps| caps.extract().1)
+        .map(|[dll_name]| dll_name)
+        .filter(|dll_name| *dll_name != "kernel32")
+        .collect();
+
+    if !dll_names.is_empty() {
+        dll_names.sort_unstable();
+        dll_names.dedup();
+
+        for dll_name in dll_names {
+            write!(&mut f, r#"#[link(name = "{dll_name}")]"#).unwrap();
+            f.write_all("\n".as_bytes()).unwrap();
+        }
```

Comment:
While the import library is often named the same as the DLL, this isn't always true (e.g. there's a Synchronization.lib but not synchronization.dll). This doesn't affect cc-rs at the moment but it could do in the future. I wonder if it would be better to maintain a manual mapping, even if this is more verbose?

Reply:
> this isn't always true (e.g. there's a Synchronization.lib but not synchronization.dll)

How many of these cases are there?

If there're only a few, then I could hard code it here, so that people don't have to modify the `macro_rules!`

Reply:
The Windows API is sprawling so it's hard to answer definitely but it seems like there are a lot. Complicating matters further are functions like `GetFileVersionInfoExA` which is exported from version.dll and there is version.lib but the lib doesn't contain that function. I don't know how many of them there are but cases like that would need to be handled specially by matching on the function name.

Reply:
Alternatively we could bump the msrv up to 1.65 and just use raw-dylib. That would completely side-step any issues.

Reply:
> Alternatively we could bump the msrv up to 1.65 and just use raw-dylib. That would completely side-step any issues.

I'm ok with raw-dynlib, but the problem is that we can't enable it just cc-rs, it's not a feature but a cfg.

Reply:
> The Windows API is sprawling so it's hard to answer definitely but it seems like there are a lot. 

Hmmm, I think it's fine for now, it's not like we add new windows sys binding very often.

Thank you so much, and I will merge it as now.

Reply:
> I'm ok with raw-dynlib, but the problem is that we can't enable it just cc-rs, it's not a feature but a cfg.

It's not a cfg if we implement in a macro ourselves. Then it just a `#[link ...  kind="raw-dylib"]`.

Reply:
Hmmm I didn't know we could do that, would open another PR.